### PR TITLE
add entrypoint script

### PR DESCRIPTION
**Pull request by ChrisB9**

### CODE REVIEW

File: entrypoint.d/toggle-profiler-on-startup.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env sh
+
+if [ "${PHP_DEBUGGER}" = 'xdebug' ]; then
```
Comment (Kanti) [inline entrypoint.d/toggle-profiler-on-startup.sh]:
```suggestion
if [[ ! -z "${PHP_DEBUGGER}" ]] && [[ "$PHP_DEBUGGER" != 'xdebug' ]]; then
```

Comment (ChrisB9) [reply entrypoint.d/toggle-profiler-on-startup.sh]: https://explainshell.com/explain?cmd=if+%5B%5B+%21+-z+%22%24%7BPHP_DEBUGGER%7D%22+%5D%5D+%26%26+%5B%5B+%22%24PHP_DEBUGGER%22+%21%3D+%27xdebug%27+%5D%5D%3B+then

Comment (Kanti) [reply entrypoint.d/toggle-profiler-on-startup.sh]: how about now?

Comment (ChrisB9) [reply entrypoint.d/toggle-profiler-on-startup.sh]: now it should work ;)

File: README.md [hunk 2]
```diff
@@ -48,27 +48,27 @@ services:
 
     volumes:
       - .:/app
-      - /var/run/docker.sock:/var/run/docker.sock:ro
+      - /var/run/docker.sock:/var/run/docker.sock:ro # this is optional if no node container is needed
```
Comment (Kanti) [inline README.md]:
```suggestion
# the docker socket is optional if no node container is needed
      - /var/run/docker.sock:/var/run/docker.sock:ro
```

File: docs/xdebug.md [hunk 3]
```diff
@@ -1,19 +1,25 @@
 # Xdebug
 
-You can set the default value for the PHP_DEBUGGER int your `.env` file:
+You can set the default value for the PHP_DEBUGGER in your `.env` file:
 ````env
 PHP_DEBUGGER=xdebug
```
Comment (Kanti) [inline docs/xdebug.md]:
```suggestion
```

File: docs/xdebug.md [hunk 4]
```diff
@@ -1,19 +1,25 @@
...
 XDEBUG_REMOTE_HOST=10.50.1.223
 ````
-Where `XDEBUG_REMOTE_HOST` is the ip of your IDE.
+
+`XDEBUG_REMOTE_HOST` is the ip-address of your IDE <br />
+Information on how to set up xdebug with PHPStorm is here: [Creating a PHP Debug Server](https://www.jetbrains.com/help/phpstorm/creating-a-php-debug-server-configuration.html)
+
+hint: if `PHP_DEBUGGER` is set to `xdebug`, then the xh-profiler won't profile anything
```
Comment (Kanti) [inline docs/xdebug.md]: this info should be at the xh-profiler docs

File: docs/xdebug.md [hunk 5]
```diff
@@ -1,19 +1,25 @@
...
+`XDEBUG_REMOTE_HOST` is the ip-address of your IDE <br />
+Information on how to set up xdebug with PHPStorm is here: [Creating a PHP Debug Server](https://www.jetbrains.com/help/phpstorm/creating-a-php-debug-server-configuration.html)
+
+hint: if `PHP_DEBUGGER` is set to `xdebug`, then the xh-profiler won't profile anything 
 
 ## enable and disable xdebug in running container:
 
 ````bash
-# enables xdebug and restarts fpm
+# enables xdebug and restarts fpm (hint: this also enables the xh-profiler)
```
Comment (Kanti) [inline docs/xdebug.md]:
```suggestion
# enables xdebug and restarts fpm (hint: this also disables the xh-profiler)
```

File: docs/xdebug.md [hunk 6]
```diff
@@ -1,19 +1,25 @@
...
 xdebug-enable
-# disable xdebug and restarts fpm
+# disable xdebug and restarts fpm (hint: this also enables the xh-profiler)
 xdebug-disable
+# the following alias toggles xdebug
```
Comment (Kanti) [inline docs/xdebug.md]:
```suggestion
# the following alias toggles xdebug and xh-profiler
```

File: run_tests.sh [hunk 7]
```diff
@@ -2,8 +2,8 @@
 
 # php should be callable
 php -v &&
-# php should have speficic modules installed
-php -m | grep xdebug || exit 1 &&
+# php should have specific modules installed
+xdebug-enable && php -m | grep xdebug || exit 1 &&
```
Comment (Kanti) [inline run_tests.sh]:
```suggestion
xdebug-enable && php -m | grep xdebug || exit 1 &&
xdebug-disable && php -m | grep -v xdebug || exit 1 &&
```

File: docs/xdebug.md (hunk 4 above)
Comment (ChrisB9) [reply docs/xdebug.md]: good point

File: docs/profiling.md [hunk 8]
```diff
@@ -9,6 +9,8 @@ Profiling is active by default. The recording can be triggered in to different w
   
 That should flood your xhgui with requests.
 
+hint: if `PHP_DEBUGGER` is set to `xdebug`, then the xh-profiler won't profile anything
```
Comment (Kanti) [inline docs/profiling.md]:
```suggestion
hint: if xdebug is enabled, then the xh-profiler won't profile anything 
```

Comment (ChrisB9) [reply docs/profiling.md]: i can have both xdebug and the profiler active at the same time. xhprof is disabled through the env-var and not xdebug in itself

Comment (ChrisB9) [inline docs/profiling.md]:
```suggestion
hint: if `PHP_DEBUGGER` is set to `xdebug` or xdebug was enabled through xdebug-enable, then the xh-profiler won't profile anything 
```

**Pull request merged**